sdk/tables/azure-data-tables/src/table_clients.cpp

```diff
@@ -449,7 +450,7 @@ Azure::Response<Models::AddEntityResult> TableClient::AddEntity(
 {
   (void)options;
```

Comment:
Unrelated to this PR: Why do we accept an `AddEntityOptions` param here, if we are not going to use it for anything?

.NET doesn't expose this:
https://github.com/Azure/azure-sdk-for-net/blob/4e1173308d32f0413788359a87376e673f2ab15b/sdk/tables/Azure.Data.Tables/src/TableClient.cs#L603

GoLang has it, but it contains MetadataFormat, rather than it being empty/inheriting from UpsertOptions:
https://github.com/Azure/azure-sdk-for-go/blob/3ebd0d439f9d8aa06a0764a60892a8001b997a34/sdk/data/aztables/client.go#L247

Reply:
https://github.com/Azure/azure-sdk-for-cpp/issues/6231